src/fsharp/TypedTreePickle.fs

```diff
@@ -2512,6 +2512,7 @@ and p_op x st =
     (* 30: TOp.TupleFieldGet  when evalTupInfoIsStruct tupInfo = true *)
     | TOp.AnonRecd info   -> p_byte 31 st; p_anonInfo info st
     | TOp.AnonRecdGet (info, n)   -> p_byte 32 st; p_anonInfo info st; p_int n st
+    | TOp.UnionCaseTest a         -> p_byte 33 st; p_ucref a st
```

Comment:
Just to mention this is going to be a problematic breaking change, making code compiled with this feature on unconsumable by previous F# compilers.  There is no specific guard against this in previous compilers.  

I'll need to take a closer look at this to work out what to do about it.

Reply:
I think we should encode/decode using existing constructs, specifically a `DecisionTreeTest.UnionCase` with true/false results.  These are correctly compiled in IlxGen.fs (or should be).

Reply:
Yeah that seems to be a better implementation overall. Thanks for the tip!